### PR TITLE
Hota h3m support

### PR DESCRIPTION
**Pull request by IvanSavenko**
Final PR, goal is to make all default hota maps playable in vcmi. 

For testing make sure to get changes from my PR in hota repository.

Known issues:
- [x] **Beware of Demons**: Requires support of victory condition "Survive beyond a time limit"
- [x] **Beware of Demons**: Requires support of placing hero directly in boat
- [x] **Trial of Flame**: Requires support of victory condition "Defeat all monsters"
- [x] **Dead Man's Tales**: Loads in-game, but customized heroes have broken portraits.
- [x] On-map monsters are placed one tile off their actual location
- [x] Check loading of hota h3m via vcmi map editor
- [x] Cove - special buildings are not loaded if added in map editor

Unrelated discovered issues:
- [x] Adventure map tavern - adventure map cursor stays, should be changed to pointer
- [ ] Town Gates (map object) - not implemented
- [ ] Junkman - not implemented
- [ ] Seafaring Academy - not implemented
- [ ] Water Shrines - should use "blockVisit" mode, similar to other water objects
- [ ] Water creature banks - should use "blockVisit" mode, 
- [x] Some static hota water objects have corrupted palette
- [ ] Water version of redwood observatory - should use "blockVisit" mode
- [x] Grave - missing visit text
- [ ] External Shipyards - should generate new ship type
- [x] Shrine (Beware of Demons map) - text placeholder for spell name was not replaced

To be done separately:
- [ ] New quests for seer hut / quest guards are not implemented
- [ ] Creature split dialog - hotkeys don't work (Enter, Escape, left/right)

### CODE REVIEW

**Comment by IvanSavenko** [comment]
@dydzio0614 
Yeah, I known about them. Code for them is in here:
https://github.com/vcmi/vcmi/blob/develop/config/mapOverrides.json#L819

Will think as to how to integrate them into hota support. In H3 these victory conditions are not part of map format but are actually hardcoded in exe. While in hota they are in h3m

**Comment by IvanSavenko** [comment]
Presumably complete, remaining points are mostly about map objects not implemented in hota mod, and don't require in-code changes.

Will update & test hota mod to implement these objects / fix these issues, but C++ part is likely done.

File: Mods/vcmi/config/vcmi/english.json [hunk 1]
```diff
@@ -169,6 +169,15 @@
...
 	"vcmi.randomMapTab.widgets.roadTypesLabel"       : "Road Types",
 	
+	// Custom victory conditions for H3 campaigns and HotA maps
+	"vcmi.map.victoryCondition.daysPassed.toOthers" : "Ворогу вдалося вижити до сьогоднішнього дня. Він переміг!",
+	"vcmi.map.victoryCondition.daysPassed.toSelf" : "Вітаємо! Вам вдалося залишитися в живих. Перемога за вами!",
+	"vcmi.map.victoryCondition.eliminateMonsters.toOthers" : "Ворог переміг усіх звірів, що заполонили цю землю, і святкує перемогу!",
+	"vcmi.map.victoryCondition.eliminateMonsters.toSelf" : "Вітаємо! Ви перемогли усіх звірів, що заполонили цю землю, і можете святкувати перемогу!",
+	"vcmi.map.victoryCondition.collectArtifacts.message" : "Здобути три артефакти",
+	"vcmi.map.victoryCondition.angelicAlliance.toSelf" : "Вітаємо! Усі ваші вороги переможені, і ви маєте Альянс Ангелів! Перемога ваша!",
+	"vcmi.map.victoryCondition.angelicAlliance.message" : "Перемогти всіх ворогів і створити Альянс Ангелів",
```
Comment (SoundSSGood) [inline Mods/vcmi/config/vcmi/english.json:179]: Is it correct translation for this file?

Comment (IvanSavenko) [reply Mods/vcmi/config/vcmi/english.json:179]: Ugh... I swapped EN and UA translations...
Will fix. Thanks for noticing

**Comment by Nordsoft91** [comment]
Town gate and seafaring academy are actually implemented, I will update configs in hota mod later

**Pull request merged**